### DISCUSSION TITLE
显示类型参数声明来指定类型参数

### DISCUSSION THREAD

**Discussion by ZongYangL** [krahets/hello-algo#620] [Q&A]
匿名内部类创建泛型实例,显示类型参数声明来指定类型参数，且使用双括号初始化的匿名内部类可能会导致一些性能问题，最好还是分开
1、chapter_tree/binary_tree_bfs/binary_tree_bfs 类中 的 
Queue<TreeNode> queue = new LinkedList<>() {{ add(root); }}; 
需要改为
Queue<TreeNode> queue = new LinkedList<TreeNode>() {{ add(root); }}; 

2、 chapter_graph/graph_bfs类中的
Set<Vertex> visited = new HashSet<>() {{ add(startVet); }};
需要改为
Set<Vertex> visited = new HashSet<Vertex>() {{ add(startVet); }};

3、utils/TreeNode类中的
Queue<TreeNode> queue = new LinkedList<>() {{ add(root); }};
需要改为
Queue<TreeNode> queue = new LinkedList<TreeNode>() {{ add(root); }};

**Comment by krahets** [comment]
Hi，如果想要更新代码，请提交 Pull Request

**Reply by ZongYangL** [reply]
ok, 我在创建 Pull Request时无法向仓库添加审查者，可以授予下写入权限吗

**Reply by krahets** [reply]
创建 PR 不需要权限，可以参考 https://www.hello-algo.com/chapter_appendix/contribution/

**Reply by ZongYangL** [reply]
<img width="875" alt="image" src="https://github.com/krahets/hello-algo/assets/40024347/d297ffaa-220b-4f1f-85c0-5ff32088d268">
已提交Pull Request，看这里提示是需要添加reviewer
